NineChronicles.Headless/GraphTypes/StandaloneQuery.cs

```diff
@@ -458,6 +458,52 @@ TransferNCGHistory ToTransferNCGHistory(TxSuccess txSuccess, string? memo)
                 name: "addressQuery",
                 description: "Query to get derived address.",
                 resolve: context => new AddressQuery(standaloneContext));
+            Field<NonNullGraphType<BooleanGraphType>>(
+                name: "replayBlocks",
+                arguments: new QueryArguments(
+                    new QueryArgument<LongGraphType>
+                    {
+                        Name = "startIndex"
+                    },
+                    new QueryArgument<LongGraphType>
+                    {
+                        Name = "endIndex"
+                    },
+                    new QueryArgument<IntGraphType>
+                    {
+                        Name = "repeatCount",
+                        DefaultValue = 1
+                    }
+                ),
+                resolve: context =>
+                {
+                    var startIndex = context.GetArgument<long?>("startIndex");
+                    var endIndex = context.GetArgument<long?>("endIndex");
+                    var repeatCount = context.GetArgument<int>("repeatCount");
+                    if (!(standaloneContext.BlockChain is { } chain))
+                    {
+                        return null;
+                    }
+
+                    var store = standaloneContext.Store;
+                    var stateStore = standaloneContext.NineChroniclesNodeService!.NodeService.StateStore;
+                    var blockChain = new BlockChain<NCAction>(chain.Policy, chain.StagePolicy, store, stateStore, chain.Genesis);
```

Comment:
It seems there is already `BlockChain<T>`, `chain`. Why does it create a new `BlockChain<T>` instance?